network/server/tchannelthrift/node/service.go

```diff
@@ -350,7 +350,8 @@ func (s *service) FetchBlocksMetadataRaw(tctx thrift.Context, req *rpc.FetchBloc
 
 	for _, fetchedMetadata := range fetchedResults {
 		blocksMetadata := s.blocksMetadataPool.Get()
-		blocksMetadata.ID = fetchedMetadata.ID.Data()
+		// LEAK(@kobolog): `fetched` is closed before return, can ID expire?
+		blocksMetadata.ID = append([]byte{}, fetchedMetadata.ID.Data()...)
```

Comment:
So what might be best in terms of efficiency, you can `ctx.RegisterCloser(fetched)` then that way the ID won't expire.